### PR TITLE
Angular 12 upgrade epic

### PR DESCRIPTION
**Pull request by dunqan**

### CODE REVIEW

File: tools/build-lib/augmented-types/index.js [hunk 1]
```diff
@@ -38,7 +38,7 @@ function augmentedTypesBuilder(options, context) {
  */
 function ngPackagrBuild(context, options) {
     return __awaiter(this, void 0, void 0, function* () {
-        const builderRun = yield context.scheduleBuilder('@angular-devkit/build-ng-packagr:build', options);
+        const builderRun = yield context.scheduleBuilder('@angular-devkit/build-angular:ng-packagr', options);
```
Comment (Platonn) [inline tools/build-lib/augmented-types/index.js:41]: nitpick: shall we update also the name of builder in the code comments:
image

File: feature-libs/storefinder/core/facade/store-finder.service.ts [hunk 2]
```diff
@@ -108,7 +108,7 @@ export class StoreFinderService implements OnDestroy {
       this.clearWatchGeolocation(new StoreFinderActions.FindStoresOnHold());
       this.geolocationWatchId = this.winRef.nativeWindow.navigator.geolocation.watchPosition(
         // TODO: Replace to GeolocationPosition when updating to new TS version
```
Comment (marlass) [inline feature-libs/storefinder/core/facade/store-finder.service.ts]:
```suggestion
```

File: projects/schematics/src/add-spartacus/index.ts [hunk 3]
```diff
@@ -248,7 +245,7 @@ export function addSpartacus(options: SpartacusOptions): Rule {
 
       updateAppModule(options.project),
       installStyles(options),
-      updateMainComponent(project, options),
+      updateMainComponent(project as WorkspaceProject, options),
```
Comment (marlass) [inline projects/schematics/src/add-spartacus/index.ts]: Why this `as` ?

File: feature-libs/cart/tsconfig.lib.json [hunk 4]
```diff
@@ -22,8 +23,7 @@
     "strictMetadataEmit": true,
     "fullTemplateTypeCheck": true,
     "strictInjectionParameters": true,
-    "enableResourceInlining": true,
-    "strictTemplates": true
```
Comment (Platonn) [inline feature-libs/cart/tsconfig.lib.json:26]: Why did we remove strctTemplates option?

Same question for `feature-libs/checkout/tsconfig.lib.json`

File: projects/schematics/src/add-spartacus/index.ts (hunk 3 above)
Comment (dunqan) [reply projects/schematics/src/add-spartacus/index.ts]: Good catch, removed!

File: projects/core/src/routing/facade/routing.service.ts [hunk 5]
```diff
@@ -121,7 +125,7 @@ export class RoutingService {
    * @returns Promise that resolves to `true` when navigation succeeds,
    *          to `false` when navigation fails, or is rejected on error.
    */
-  goByUrl(url: string, extras?: NavigationExtras): Promise<boolean> {
+  goByUrl(url: string, extras?: NavigationBehaviorOptions): Promise<boolean> {
```
Comment (Platonn) [inline projects/core/src/routing/facade/routing.service.ts:128]: 👍

**Comment by Platonn** [comment]
When running `ng serve` (for developement), now the app builds much slowee than before, on start and on each code change. Also the warning is presented (as used to be only for ng serve --prod):
image

It seems to be a known issue:
https://stackoverflow.com/questions/67598130/angular-12-ng-serve-builds-apps-slowly-almost-like-production-builds
https://github.com/angular/angular-cli/issues/20801#issuecomment-868324594

Ng team says it has been improved in v12.1. Maybe worth checking? 
https://github.com/angular/angular-cli/issues/20801#issuecomment-868324594

**Review by Platonn — DISMISSED** [review]

**Review by Platonn — DISMISSED** [review]

**Review by Platonn — APPROVED** [review]

**Review by Platonn — APPROVED** [review]

**Review by Platonn — APPROVED** [review]

**Pull request merged**